File: configlibrary/src/main/java/com/shmedo/configlibrary/ble/enums/OsmometerStatus.java
package com.shmedo.configlibrary.ble.enums;

/**
 * Created by adu on 2018/1/8.
 * 渗压计功能状态
 */
public enum OsmometerStatus {

    OSMOMETER_OPEN(1), OSMOMETER_CLOSE(2);
    //1、开启      2、关闭
    private int status;

    OsmometerStatus(int i) {
        this.status = i;
    }

    public int toInt() {
        return status;
    }

    public static OsmometerStatus valueOf(int status) {
        switch (status) {
            case 1:
                return OSMOMETER_OPEN;
            case 2:
                return OSMOMETER_CLOSE;
            default:
                return OSMOMETER_CLOSE;
        }
    }
}

File: lib_device_base/src/main/java/com/shmedo/lib/device/base/md_cmd/enums/OsmometerStatus.kt
package com.shmedo.lib.device.base.md_cmd.enums

/**
 * Created by adu on 2018/1/8.
 * 渗压计功能状态
 */
enum class OsmometerStatus(//1、开启      2、关闭
    private val status: Int
) {
    OSMOMETER_OPEN(1),
    OSMOMETER_CLOSE(2);

    fun toInt(): Int {
        return status
    }

    companion object {
        fun value(status: Int): OsmometerStatus {
            return when (status) {
                1 -> OSMOMETER_OPEN
                2 -> OSMOMETER_CLOSE
                else -> OSMOMETER_CLOSE
            }
        }
    }
}
